### ISSUE TITLE
Nested/Child bookmarks not changed

### ISSUE THREAD

**Issue by az111** [beatngu13/pdf-zoom-wizard#50]
Hi,

I've noticed that the bookmark changes do not get applied to nested/child bookmarks within documents. 

Thanks

**Comment by beatngu13** [comment]
Hi @az111,

Could you please use the [issue template for bug reports](https://github.com/beatngu13/pdf-zoom-wizard/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) and complete the missing information?

Unfortunately, I cannot reproduce the bug. Nested/child bookmarks get modified with my samples. Could you provide a PDF that leads to the error on your machine?

**Comment by az111** [comment]
Thanks for your quick response

To Reproduce Steps to reproduce the behavior:

1. Using: [00024572.pdf](https://github.com/beatngu13/pdf-zoom-wizard/files/3108180/00024572.pdf)
2. Try to change bookmarks from 'Inherit Zoom' as in original to 'Fit Page'
3. Only top level bookmarks are changed, when either modifying original or making a copy 


Desktop (please complete the following information):

    OS: [Windows 10 64-bit]
    Java version: [8]
    PDF Zoom Wizard version: [0.40]

**Comment by beatngu13** [comment]
From the log file:

```
17:10:29.489 [JavaFX Application Thread] INFO  Launcher - Launching PDF Zoom Wizard v0.4.0.
17:10:46.491 [Thread-5] INFO  Wizard - Start working on '/Users/kruse/Downloads/pdfzoomwizard/test-data/00024572.pdf'. PDF document(s) will be saved with serialization mode 'Incremental'.
17:10:46.491 [Thread-5] INFO  Wizard - Processing '00024572.pdf'.
17:10:46.574 [Thread-5] INFO  Wizard - Modified bookmark 'PRODUCT MONOGRAPH' to use mode 'Fit' and zoom 'null'.
17:10:46.574 [Thread-5] INFO  Wizard - Modified bookmark 'Table of Contents' to use mode 'Fit' and zoom 'null'.
17:10:46.575 [Thread-5] INFO  Wizard - Modified bookmark 'PART  I: HEALTH PROFESSIONAL INFORMATION' to use mode 'Fit' and zoom 'null'.
17:10:46.575 [Thread-5] INFO  Wizard - Modified bookmark 'PART II:  SCIENTIFIC INFORMATION' to use mode 'Fit' and zoom 'null'.
17:10:46.575 [Thread-5] INFO  Wizard - Modified bookmark 'IMPORTANT: PLEASE READ' to use mode 'Fit' and zoom 'null'.
17:10:46.575 [Thread-5] INFO  Wizard - Modified bookmark 'PART III: CONSUMER INFORMATION' to use mode 'Fit' and zoom 'null'.
17:10:46.592 [Thread-5] INFO  Wizard - Modified 6 bookmark(s) in '00024572.pdf'.
17:10:46.592 [Thread-5] INFO  Wizard - Modified 6 bookmark(s) in 1 file(s).
```

Indeed, only the top-level bookmarks get modified. This is not the case with my samples. Good catch!

Will look into this.

**Comment by beatngu13** [comment]
Found the issue. I only modified nested bookmarks if the child count was greater than zero. But according to the PDF Reference (`v1.6`, section 8.2.2), the count might be negative if the corresponding bookmark is closed.

I'm already working on a fix which I will release soon (see #51).

**Comment by az111** [comment]
Thanks!

**Comment by beatngu13** [comment]
Released [v0.4.1](https://github.com/beatngu13/pdf-zoom-wizard/releases/tag/v0.4.1). Would you mind to give it a try and let me know if it works for you?
